### DISCUSSION TITLE
Reuse workflow: dynamic branch reference (github.ref_name)

### DISCUSSION THREAD

**Discussion by hocinemala** [community/community#24661] [Apps, API and Webhooks]
<p>Hello,</p>
<p>When using reuse workflow, we need to mention the branch to reference the yml workflow.</p>
<p>is working:<br>
uses: Company/myRepo/.github/workflows/wf.yml@main</p>
<p>This is not working when I use the predefined variable (github.ref_name) to dynamcally ref the branch:</p>
<p>uses: Company/myRepo/.github/workflows/wf.yml@${{github.ref_name}}</p>
<p>Is someone tried to make this working ?</p>
<p>Regards</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Technically yes, the GitHub team is working on making it possible to use reusable workflow files from the current branch:</p><aside class="quote quote-modified" data-post="30" data-topic="203690">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/ref-head-in-reusable-workflows/203690/30">Ref Head in Reusable Workflows</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    Hey everyone - I’m the product manager for reusable workflows. We’re currently working on the local reference scenario. It should be available in the next couple of weeks. 
Here’s how it will work: 
If a reusable workflow is located in the same repository as the caller, you can reference it with a local path:  {path}/{filename}. We’ll automatically use the current ref. 
Here’s an example that shows the two ways you’ll have to reference a reusable workflow: 
jobs:
  call-workflow-1:
    uses: .gi…
  </blockquote>
</aside>

<p>Which of course implies it’s not possible right now. :sweat_smile:</p>

**Comment by asingh2023** [comment]
Hi Github Team,


Is there any update on this particular feature in GitHub action related to dynamic branch referencing.



Thanks in advance.

**Reply by airtower-luna** [reply]
Calling a workflow *in the current repository* is possible without specifying a ref now, see: https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow
